pallets/ajuna-awesome-avatars/src/lib.rs

```diff
@@ -402,6 +375,51 @@ pub mod pallet {
 				.collect::<Vec<_>>();
 			Dna::try_from(dna).map_err(|_| Error::<T>::IncorrectDna.into())
 		}
+
+		pub(crate) fn do_mint(player: &T::AccountId, how_many: MintCount) -> DispatchResult {
+			ensure!(Self::mint_available(), Error::<T>::MintUnavailable);
+
+			let current_block = <frame_system::Pallet<T>>::block_number();
+			if let Some(last_block) = Self::last_minted_block_numbers(&player) {
+				let cooldown = Self::mint_cooldown();
+				ensure!(current_block > last_block + cooldown, Error::<T>::MintCooldown);
+			}
+
+			let how_many = how_many as usize;
+			let max_ownership = (MAX_AVATARS_PER_PLAYER as usize)
+				.checked_sub(how_many)
+				.ok_or(ArithmeticError::Underflow)?;
+			ensure!(Self::owners(&player).len() <= max_ownership, Error::<T>::MaxOwnershipReached);
+
+			let active_season_id = Self::active_season_id().ok_or(Error::<T>::OutOfSeason)?;
+			let active_season = Self::seasons(active_season_id).ok_or(Error::<T>::UnknownSeason)?;
+
+			let generated_avatars = (0..how_many)
+				.map(|_| {
+					let dna = Self::random_dna(player, &active_season)?;
+					let avatar = Avatar { season: active_season_id, dna };
+					let avatar_id = T::Hashing::hash_of(&avatar);
+					Ok((avatar_id, avatar))
+				})
+				.collect::<Result<Vec<(AvatarIdOf<T>, Avatar)>, DispatchError>>()?;
+
+			Owners::<T>::try_mutate(&player, |avatar_ids| -> DispatchResult {
+				let generated_avatars_ids = generated_avatars
+					.into_iter()
+					.map(|(avatar_id, avatar)| {
+						Avatars::<T>::insert(avatar_id, (&player, avatar));
+						avatar_id
+					})
+					.collect::<Vec<_>>();
+				ensure!(
+					avatar_ids.try_extend(generated_avatars_ids.clone().into_iter()).is_ok(),
+					Error::<T>::MaxOwnershipReached
```

Comment:
Since max ownership is already checked upfront, `try_extend` will never fail. The ensure here is used just to avoid using `unwrap()`.